config/main.py

```diff
@@ -4413,6 +4547,123 @@ def delete(ctx):
     config_db.set_entry('SFLOW', 'global', sflow_tbl['global'])
 
+#
+# set ipv6 link local mode on a given interface
+#
+def set_ipv6_link_local_only_on_interface(config_db, interface_dict, interface_type, interface_name, mode):
+
+    curr_mode = config_db.get_entry(interface_type, interface_name).get('ipv6_use_link_local_only')
+    if curr_mode is not None:
+        if curr_mode == mode:
+            return
+    else:
+        if mode == "disable":
+            return
+
+    if mode == "enable":
+        config_db.mod_entry(interface_type, interface_name, {"ipv6_use_link_local_only": mode})
+        return
+
+    # If we are disabling the ipv6 link local on an interface, and if no other interface
+    # attributes/ip addresses are configured on the interface, delete the interface from the interface table
+    exists = False
+    for key in interface_dict.keys():
+        if not isinstance(key, tuple):
+            if interface_name == key:
+                #Interface bound to non-default-vrf do not delete the entry
+                if 'vrf_name' in interface_dict[key]:
+                    if len(interface_dict[key]['vrf_name']) > 0:
+                        exists = True
+                        break
+            continue
+        if interface_name in key:
+            exists = True
+            break
+
+    if exists:
+        config_db.mod_entry(interface_type, interface_name, {"ipv6_use_link_local_only": mode})
+    else:
+        config_db.set_entry(interface_type, interface_name, None)
+
+#
+# 'ipv6' group ('config ipv6 ...')
+#
+
+@config.group()
+@click.pass_context
+def ipv6(ctx):
+    """IPv6 configuration"""
+
+#
+# 'enable' command ('config ipv6 enable ...')
+#
+@ipv6.group()
+@click.pass_context
+def enable(ctx):
+    """Enable IPv6 on all interfaces """
+
+#
+# 'link-local' command ('config ipv6 enable link-local')
+#
+@enable.command('link-local')
+@click.pass_context
+def enable_link_local(ctx):
+    """Enable IPv6 link-local on all interfaces """
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    vlan_member_table = config_db.get_table('VLAN_MEMBER')
+    portchannel_member_table = config_db.get_table('PORTCHANNEL_MEMBER')
+
+    mode = "enable"
+
+    # Enable ipv6 link local on VLANs
+    vlan_dict = config_db.get_table('VLAN')
+    for key in vlan_dict.keys():
+        set_ipv6_link_local_only_on_interface(config_db, vlan_dict, 'VLAN_INTERFACE', key, mode)
+
+    # Enable ipv6 link local on PortChannels
+    portchannel_dict = config_db.get_table('PORTCHANNEL')
+    for key in portchannel_dict.keys():
+        if interface_is_in_vlan(vlan_member_table, key):
+            continue
+        set_ipv6_link_local_only_on_interface(config_db, portchannel_dict, 'PORTCHANNEL_INTERFACE', key, mode)
+
+    port_dict = config_db.get_table('PORT')
+    for key in port_dict.keys():
+        if interface_is_in_portchannel(portchannel_member_table, key) or interface_is_in_vlan(vlan_member_table, key):
```

Comment:
What if we break-out the port after this global config? how do we apply this config to the new ports? I think, this global config implementation could be done better as per the HLD to have global field in the device_metadata, please re-visit.

Reply:
As per HLD, the global config will enable/disable the link local mode on all eligible interfaces. 
If we does the break-out after the global config, we have to apply the global config again or interface config on the new ports.

Global field in device_metadata, not planned as part this PR, it should be an future item.

Reply:
Please create the issue for tracking.

Reply:
Here is the issue - https://github.com/Azure/sonic-buildimage/issues/8364